packages/fleather/lib/src/rendering/embed_proxy.dart

```diff
@@ -50,3 +50,93 @@ class RenderEmbedProxy extends RenderProxyBox implements RenderContentProxyBox {
   @override
   double get preferredLineHeight => size.height;
 }
+
+/// Proxy to an arbitrary embeddable [RenderBox].
+///
+/// Computes necessary editing metrics based on the dimensions of the childlren
+/// render boxes.
+class RenderGroupEmbedProxy extends RenderFlex
+    implements RenderContentProxyBox {
+  RenderGroupEmbedProxy({
+    TextDirection? textDirection,
+  }) : super(textDirection: textDirection);
+
+  @override
+  List<TextBox> getBoxesForSelection(TextSelection selection) {
+    return [];
+  }
+
+  @override
+  double? getFullHeightForCaret(TextPosition position) {
+    return preferredLineHeight;
+  }
+
+  @override
+  Offset getOffsetForCaret(TextPosition position, Rect caretPrototype) {
+    if (childCount == 0 || position.offset == 0) {
+      return Offset.zero;
+    }
+
+    RenderBox? child;
+    double accWidth = 0;
+
+    for (var i = 0; i < childCount; i += 1) {
+      if (position.offset <= i) {
+        return Offset(accWidth, 0.0);
+      }
+
+      if (i == 0) {
+        child = firstChild!;
+      } else {
+        child = childAfter(child!)!;
+      }
+
+      // TODO: Can we use `semanticBounds` here?
+      // As an alternative I can also calculate the size myself:
+      //   final width = (child as RenderBox).getMinIntrinsicWidth(size.height);
+      final width = child.semanticBounds.width;
```

Comment:
How do I get child's width correctly?

Reply:
can we use something like:
```
child = children[position]; 
return child.localToGlobal(Offset(child.size.with, 0), this);
```
to get the offset of a child directly? (without needed a for loop)